### PR TITLE
Remove experimental label from apollo usage reporting configs

### PR DESCRIPTION
**Pull request by bonnici**
Removes the experimental label and renames some config options related to Apollo usage reporting:
* `experimental_apollo_metrics_generation_mode` has been removed since the Rust implementation has been the default since v1.49.0 and it is generating reports identical to the router-bridge implementation
* `telemetry.apollo.experimental_apollo_signature_normalization_algorithm` is now `telemetry.apollo.signature_normalization_algorithm`
* `telemetry.apollo.experimental_apollo_metrics_reference_mode` is now `telemetry.apollo.metrics_reference_mode`

The defaults have not been changed.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

[ROUTER-696]: https://apollographql.atlassian.net/browse/ROUTER-696?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by goto-bus-stop** [comment]
Hm, shouldn't these eventually be removed and only the "new" and "enhanced" modes be available?

**Comment by goto-bus-stop** [comment]
We may want to remove `legacy` in 1.x for LTS too, I think, since it requires JS.

**Comment by goto-bus-stop** [comment]
Or coming at it from a different angle--the `metrics_generation_mode: legacy` uses the old JavaScript implementation. If the Rust implementation works well (as it appears to do), what's the point in continuing to support the `legacy` mode?

File: .changesets/config_apollo_telemetry_config_rename.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+###  ([#5807](https://github.com/apollographql/router/pull/5807))
+
+All known issues related to the new Apollo usage report generation have been resolved so we are renaming some experimental options to be non-experimental.
+* `experimental_apollo_metrics_generation_mode` is now `apollo_metrics_generation_mode`
```
Comment (goto-bus-stop) [inline .changesets/config_apollo_telemetry_config_rename.md:4]:
```suggestion
* `experimental_apollo_metrics_generation_mode` is removed
```

Comment (bonnici) [reply .changesets/config_apollo_telemetry_config_rename.md:4]: Oops thanks, this must have been a copy-paste error. I had something else to put here.

**Comment by bonnici** [comment]
@goto-bus-stop this should be good to go now, can you take a look so that I can get it into the 1.54 release?

File: .changesets/config_apollo_telemetry_config_rename.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+###  ([#5807](https://github.com/apollographql/router/pull/5807))
```
Comment (bnjjj) [inline .changesets/config_apollo_telemetry_config_rename.md:1]:
```suggestion
###  Remove experimental label from apollo usage reporting configs ([#5807](https://github.com/apollographql/router/pull/5807))
```

**Review by bnjjj — DISMISSED** [review]

**Review by lrlna — APPROVED** [review]
once the title is added to the changeset, it looks good to me! I am glad we are getting rid of one other thing that relies on JS!

File: .changesets/config_apollo_telemetry_config_rename.md (hunk 2 above)
Comment (bonnici) [reply .changesets/config_apollo_telemetry_config_rename.md]: Thanks! I missed that.

**Pull request merged**